### PR TITLE
ETV-2300-new-textarea-component

### PR DESCRIPTION
**Pull request by elenawq**
### [Jira]: https://etvas.atlassian.net/browse/ETV-2300

Screenshot 2020-08-31 at 17 35 03

### CODE REVIEW

File: src/TextArea/TextArea.jsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    PropTypes.string,
+    PropTypes.number,
+    PropTypes.object
+  ])
+}
+
+TextArea.defaultProps = {
+  autoFocus: false,
+  rows: 5,
+  allowResize: false,
```
Comment (alsofronie) [inline src/TextArea/TextArea.jsx:176]: Default HTML behaviour is to allow resize.

```suggestion
  allowResize: true,
```

File: src/TextArea/TextArea.jsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  ]),
+  icLeft: PropTypes.string,
+  icRight: PropTypes.string,
+  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+  label: PropTypes.node,
+  name: PropTypes.string,
+  /** dummy space added for consistent spacing with validated inputs.
+   *
+   * remove space by setting this to true */
+  noBottomSpace: PropTypes.bool,
```
[the remark this answers is not in the document]
Comment (elenawq) [reply src/TextArea/TextArea.jsx:158]: I see that it is used in the ErrorMessage to determine whether or not we should display a small space underneath the text field

[the remark this answers is not in the document]
Comment (elenawq) [reply src/TextArea/TextArea.jsx:158]: This is available for Input, should we also use it in the TextArea?

[the remark this answers is not in the document]
Comment (alsofronie) [reply src/TextArea/TextArea.jsx]: My mistake

File: src/TextArea/ErrorMessage.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  }
+
+  return null
+}
+
+const Dummy = styled.div`
+  display: ${({ hide }) => (hide ? 'none' : 'block')};
+  height: 20px;
+  opacity: 0;
+  visibility: hidden;
```
Comment (alsofronie) [inline src/TextArea/ErrorMessage.jsx:27]: Either use:

```
  return <Dummy hide={!preserveSpace} />
}

const Dummy = styled.div`
  display: ${({ hide }) => (hide ? 'none' : 'block')};
  height: 20px;
  opacity: 0;
  visibility: hidden;
```

or

```suggestion
  if (preserveSpace) {
    return <Dummy />
  }

  return null
}

const Dummy = styled.div`
  display: block;
  height: 20px;
  opacity: 0;
  visibility: hidden;
```

**Review by alsofronie — CHANGES_REQUESTED** [review]

**Review by alsofronie — APPROVED** [review]

**Pull request merged**